realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -526,19 +586,168 @@ boolean addAllInternal(Collection<? extends Byte> c) {
         return osSet.collectionFunnel(collection, OsSet.ExternalCollectionOperation.ADD_ALL);
     }
 
+    @Override
+    boolean removeAllInternal(Collection<?> c) {
+        // Collection has been type-checked from caller
+        //noinspection unchecked
+        NativeMixedCollection collection = NativeMixedCollection.newIntegerCollection((Collection<Byte>) c);
+        return osSet.collectionFunnel(collection, OsSet.ExternalCollectionOperation.REMOVE_ALL);
+    }
+
     @Override
     boolean retainAllInternal(Collection<?> c) {
+        // Collection has been type-checked from caller
         //noinspection unchecked
         NativeMixedCollection collection = NativeMixedCollection.newIntegerCollection((Collection<Byte>) c);
         return osSet.collectionFunnel(collection, OsSet.ExternalCollectionOperation.RETAIN_ALL);
     }
+}
+
+/**
+ * TODO
+ */
+class BinaryOperator extends SetValueOperator<byte[]> {
+
+    public BinaryOperator(BaseRealm baseRealm, OsSet osSet, Class<byte[]> valueClass) {
+        super(baseRealm, osSet, valueClass);
+    }
+
+    @Override
+    boolean add(@Nullable byte[] value) {
+        return osSet.add(value);
+    }
+
+    @Override
+    boolean containsInternal(@Nullable Object o) {
+        byte[] value;
+        if (o == null) {
+            value = null;
+        } else {
+            value = (byte[]) o;
+        }
+        return osSet.contains(value);
+    }
+
+    @Override
+    boolean removeInternal(@Nullable Object o) {
+        return osSet.remove((byte[]) o);
+    }
+
+    @Override
+    boolean containsAllInternal(Collection<?> c) {
+        for (Object value : c) {
```

Comment:
Wouldn't this have been checked with your new checks in the `SetValueOperator.containsAll`?

Similarly in a few other places.

Reply:
You're totally right about this 🤦